### PR TITLE
(chore): remove python 3.9

### PR DESCRIPTION
**Pull request by ilan-gold**
Towards following https://scientific-python.org/specs/spec-0000/
- [ ] Closes #
- [ ] Tests added
- [x] Release note added (or unnecessary)

### CODE REVIEW

**Review by flying-sheep — APPROVED** [review]
ah the ruff fixes are beautiful to see!

**Comment by ilan-gold** [comment]
yes @flying-sheep! we need to update `h5py` pushing now - can re-request your review or just merge

File: .azure-pipelines.yml [hunk 1]
```diff
@@ -19,13 +19,13 @@ jobs:
           RUN_COVERAGE: yes
           TEST_TYPE: "coverage"
         Python3.9:
```
Comment (flying-sheep) [inline .azure-pipelines.yml:21]: needs renaming

**Comment by flying-sheep** [comment]
> we need to update h5py pushing now

you mean update the minimum h5py version we depend on? no need to get my review, I think we have everything automated enough that except for cosmetic stuff like that test run name, it’s fine once the tests pass

**Comment by ilan-gold** [comment]
>you mean update the minimum h5py version we depend on?

yup, sounds good though. will merge once the tests pass

**Pull request merged**